website/www/site/content/en/documentation/runtime/environments.md

```diff
@@ -17,54 +17,175 @@ limitations under the License.
+
+```
+export BEAM_SDK_VERSION="2.26.0"
+export IMAGE_NAME="myrepo/mybeamsdk"
+export TAG="${BEAM_SDK_VERSION}-custom"
+
+docker tag apache/beam_python3.6_sdk "${IMAGE_NAME}:${TAG}"
+docker push "${IMAGE_NAME}:${TAG}"
+```
+
+**NOTE**: After pushing a container image, verify the remote image ID and digest matches the local image ID and digest output from `docker_images`
+
+##### Additional build parameters
+
+The docker Gradle task defines a default image repository and [tag](https://docs.docker.com/engine/reference/commandline/tag/) is the SDK version defined at [gradle.properties](https://github.com/apache/beam/blob/master/gradle.properties). The default repository is the Docker Hub `apache` namespace, and the default tag is the [SDK version](https://github.com/apache/beam/blob/master/gradle.properties) defined at gradle.properties.
+
+You can specify a different repository or tag for built images by providing parameters to the build task. For example:
+
 ```
-2. Customize the [Dockerfile](https://github.com/apache/beam/blob/master/sdks/python/container/Dockerfile). If you're adding dependencies from [PyPI](https://pypi.org/), use [`base_image_requirements.txt`](https://github.com/apache/beam/blob/master/sdks/python/container/base_image_requirements.txt) instead.
-3. [Reimage](#building-container-images) the container.
+./gradlew :sdks:python:container:py36:docker -Pdocker-repository-root="example-repo" -Pdocker-tag="2.26.0-custom"
+```
+
+builds the Python 3.6 container and tags it as `example-repo/beam_python3.6_sdk:2.26.0-custom`.
+
+From Beam 2.21.0 and later, a `docker-pull-licenses` flag was introduced to add licenses/notices for third party dependencies to the docker images. For example:
+
+```
+./gradlew :sdks:java:container:java8:docker -Pdocker-pull-licenses
+```
+creates a Java 8 SDK image with appropriate licenses in `/opt/apache/beam/third_party_licenses/`.
+
+By default, no licenses/notices are added to the docker images.
 
-### Testing customized images
 
-To test a customized image locally, run a pipeline with PortableRunner and set the `--environment_config` flag to the image path:
+## Using container images in pipelines
+
+The common method for providing a container image requires using the PortableRunner and setting the `--environment_config` flag to a given image path.
+Other runners, such as Dataflow, support specifying containers with different flags.
 
 {{< highlight class="runner-direct" >}}
+export IMAGE="my-repo/beam_python_sdk_custom"
+export TAG="X.Y.Z"
+
 python -m apache_beam.examples.wordcount \
 --input=/path/to/inputfile \
 --output /path/to/write/counts \
 --runner=PortableRunner \
 --job_endpoint=embed \
---environment_config=path/to/container/image
+--environment_config="${IMAGE}:${TAG}"
 {{< /highlight >}}
 
 {{< highlight class="runner-flink-local" >}}
+export IMAGE="my-repo/beam_python_sdk_custom"
+export TAG="X.Y.Z"
+
 # Start a Flink job server on localhost:8099
```

Comment:
It'd be nice to use `FlinkRunner` instead, which automatically starts a job server so the user doesn't have to. But it doesn't have to block this PR, I can do it in a later change.

Reply:
We are probably waiting until early next week to merge in - probably better to use the more accurate version of commands. Do you have an example of recommended arguments with FlinkRunner for Python/is that supported? I couldn't find one in the docs. 

Also, should we be doing the same things for spark? Based on a quick look it doesn't look like we have the same functionality for Spark, but I'm not very familiar with either Flink or Spark.

Reply:
It'd look something like this:

```sh
python -m apache_beam.examples.wordcount \
--input=/path/to/inputfile \
--output=/path/to/write/counts \
--runner=FlinkRunner \
--environment_type=DOCKER \
--environment_config="${IMAGE}:${TAG}"
```

And then the same for Spark, but substitute `--runner=SparkRunner`.

Reply:
edited - I took out the Flink/Spark runner server comments and code.